server/src/http/helpers/normalise-express-request.js

```diff
@@ -0,0 +1,32 @@
+function normaliseExpressRequest(request = {}) {
+  const hasRequiredProps = requestContainsExpressProperties(request);
+  
+  
+  if (request && hasRequiredProps) {
+    // Returns immutable request object
+    return Object.freeze({
+      method: request.method,
+      path: request.path,
+      headers: request.headers,
+      pathParams: request.params,
+      queryParams: request.query,
+      body: request.body
+    });
```

Comment:
The way I understand it, you're essentially locking in the parameters on the request object for the duration of the request. This is done to prevent people from maliciously changing them. Am I correct in this understanding, or is there some other reason?

Reply:
Yep that's right. Not just for potential malicious changes but to guard against other application code mutating it and causing bugs.